dev-docs/RFCs/v4.0.0/names_for_concepts_types_functions.md

```diff
@@ -0,0 +1,225 @@
+- `<func_name>Safe`
+
+The `Unsafe` version is fast, but may fail if it encounters a pentagon.
+It should return a failure code in this case.
+
+The `Safe` version is slower, but will work in all cases.
+
+The version without either suffix is intended to be the one typically
+used.
+This version will first attempt the `Unsafe` version, and if
+it detects failure, will fall back to the `Safe` version.
+Encountering pentagons is rare in most use-cases, so this version
+should usually be equivalent to the fast version, but with a guarantee
+that it will not fail.
+
+Initially, we **will not** expose the `Safe` versions to users in the API.
+We may expose them in the future if a need becomes clear.
+
+
+#### Distance
+
+| Current name |      Proposed name      |
+|--------------|-------------------------|
+| `h3Distance` | `gridDistance`          |
+| `h3Line`     | `gridPathCells`         |
+| *DNE*        | `gridPathEdges`         |
+| *DNE*        | `gridPathDirectedEdges` |
+
+
+#### Filled-In Disk With Distances
+
+|     Current name    |       Proposed name       |                Calls                 |
+|---------------------|---------------------------|--------------------------------------|
+| `hexRangeDistances` | `gridDiskDistancesUnsafe` | NONE                                 |
+| `_kRingInternal`    | `gridDiskDistancesSafe`   | NONE                                 |
+| `kRingDistances`    | `gridDiskDistances`       | `hexRangeDistances`,`_kRingInternal` |
+
+**Note**: The distances array is *optional* for `hexRangeDistances`, but *required* for the other two functions.
+
+#### Filled-In Disk Without Distances
+
+| Current name |              Proposed name               |                      Calls                       |
+|--------------|------------------------------------------|--------------------------------------------------|
+| `hexRange`   | `gridDiskUnsafe`                         | `hexRangeDistances`, does not allocate distances |
+| *DNE*        | `gridDiskSafe`                           |                                                  |
+| `kRing`      | `gridDisk`                               | `kRingDistances`, allocates and drops distances  |
+| `hexRanges`  | `gridDiskMultiUnsafe`, `gridDisksUnsafe` | N x `hexRange`                                   |
+
+- **TODO**: Do we *really* want to keep `hexRanges` in the API? It sounds like it currently has applications, but how
+  hard would it really be for a user to reproduce this functionality by just calling `hexRange` multiple times, and moving
+  the pointer before each call? It is also weird that we only provide the "unsafe" version of this function in the API.
+- **Idea**: leave it out initially, and see if anyone complains? It would be very easy to add back in.
+
+#### Hollow Ring
+
+| Current name |  Proposed name   | Calls |
+|--------------|------------------|-------|
+| `hexRing`    | `gridRingUnsafe` | NONE  |
+| *DNE*        | `gridRingSafe`   |       |
```

Comment:
Brute force approach for this - call `gridDiskDistances` and then filter out the inside rings

Reply:
Agreed. Updated it to:

```
| Current name |  Proposed name   |              Calls               |
|--------------|------------------|----------------------------------|
| `hexRing`    | `gridRingUnsafe` | NONE                             |
| *DNE*        | `gridRingSafe`   | `gridDiskDistancesSafe`          |
| *DNE*        | `gridRing`       | `gridRingUnsafe`, `gridRingSafe` |
```